.circleci/config.yml

```diff
@@ -115,23 +116,45 @@ references:
         ./cc-test-reporter format-coverage -t simplecov -o "coverage/codeclimate.$CIRCLE_NODE_INDEX.json"
         ./cc-test-reporter sum-coverage --output - coverage/codeclimate.*.json | ./cc-test-reporter upload-coverage --debug --input -
 
-  configure_aws_profile: &configure_aws_profile
+  aws_setup: &aws_setup
+    # Authenticate to AWS using OIDC v2 with the AWS CLI
+    aws-cli/setup:
+      role_arn: $ECR_ROLE_TO_ASSUME
+      region: $ECR_REGION
+
+  ecr_login: &ecr_login
     run:
-      name: Configure AWS CLI
+      name: Authenticate to the ECR repository
       command: |
-        aws configure set aws_access_key_id ${AWS_ACCESS_KEY_ID} --profile ecr-live-parliamentary-questions
-        aws configure set aws_secret_access_key ${AWS_SECRET_ACCESS_KEY} --profile ecr-live-parliamentary-questions
-        aws configure set default.region ${AWS_DEFAULT_REGION} --profile ecr-live-parliamentary-questions
+        aws ecr get-login-password --region $ECR_REGION | docker login --username AWS --password-stdin ${AWS_ECR_REGISTRY_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com
 
-  build_and_push_docker_image: &build_and_push_docker_image
+  configure_build_tag: &configure_build_tag
     run:
-      name: Build docker image
+      name: Create build tag
       command: |
-        . build.sh circleci
+        current_branch=$(git branch | grep \* | cut -d ' ' -f2)
+        short_version=$(git rev-parse --short $current_branch)
+        build_tag=cmoj-${current_branch}-${short_version}
+        echo "export BUILD_TAG=${build_tag}" >> $BASH_ENV
+        source $BASH_ENV
         mkdir -p workspace
         echo $BUILD_TAG > workspace/build_tag
+
+  build_and_push_image: &build_and_push_image
+    aws-ecr/build-image:
+      push-image: true
+      tag: $BUILD_TAG
+      region: $ECR_REGION
+      repo: $ECR_REPOSITORY
+
+  scan_for_vulnerabilities: &scan_for_vulnerabilities
+    run:
+      name: Check ECR scan results
+      command: |
+        build_tag=`cat /tmp/workspace/build_tag`
         python3 config/pipeline/aws_ecr_scan_results.py --search parliamentary-questions --tag $BUILD_TAG --slack_webhook $SLACK_WEBHOOK || true
```

Comment:
You do need to use $build_tag as that is what the variable has been called in the line above.